metrics/mobile-app-metrics.md

```diff
@@ -0,0 +1,131 @@
+
+- Provide developers an interface to query and visualize the gathered data in order to make informed decisions towards deployment, deprecation of backend services and other actions that depend on the state of an application's installed base.
+
+### Secondary Goals
+
+- Provide mobile app developers with an easy to use SDK to connect easily and securely to the metrics service
+
+### Non Goals
+
+- Identify individual devices or users in metrics events
+
+- Registering of user-defined metrics metrics
+
+## Terms
+
+For a list of terms used in this document, see [Glossary](./glossary.md).
+
+## Problem Description
+
+As mobile applications become more successful, system metrics data becomes invaluable for developers to observe the landscape of application installations and make sure further development and deployment of backend services will not affect users.
+
+## Proposed Solution
+
+A Metrics Service will be available from the Service Catalog, which can be provisioned by developers for use with their mobile applications.
+
+The extra components outlined in this proposal will be added to the existing components outlined in the [Metrics Endpoints and Auto Discovery Proposal](./mobile-app-metrics-service.md).
+
+The following diagram illustrates the components of the Metrics Service:
+
+![service diagram](./app-service-diagram.jpg)
+
+### Mobile App Client
+
+A native, platform-specific SDK for targetting the Metrics Service will be available to developers.
+
+Upon installation into a project and providing configuration for targetting the Metrics Service the SDK will automatically send general information about the general runtime for this application with a unique device ID, namely the `[SDK Version, clientID, timestamp]` tuple upon application initialization.
+
+The SDK will also instrument calls at the network level in order to gather metrics around communication and bandwith consumption.
+
+Other system resources utilization data can also be sent periodically.
+
+### Mobile Metrics Service
+
+A thin HTTP-based service will be present to receive metrics data from the mobile applications in an storage-agnostic way, then forwarding to the Storage service.
+
+Compared to mobile applications interfacing directly with the Storage service, this presents the following tradeoff:
+
+#### Pros
+
+- Possibility to upgrade or change the Storage service independently of mobile client SDK version and implementation;
+
+- Possible support for different Storage engines which can be selected based on application needs;
+
+#### Cons
+
+- Extra service deployment and scaling needs;
+
+- Development and maintenance of the new service;
+
+- Types of gathered metrics can become a subset of the underlying Storage's full range of types.
+
+<!-- TODO: mention logstash, fluentd -->
+
+### Storage
+
+The Storage service will store metrics data forwarded from the Mobile Metrics Service. Its implementation can be supplied by the following projects:
```

Comment:
I'd like to see this section be much smaller, focusing on the solution we're proposing based on any investigations etc..
It could link off to any jiras where the investigation was done, and conclusions were made.

Reply:
Agreed!

@aliok I'm going to move the contents of this section + link the google doc to the [spike issue](https://issues.jboss.org/browse/AEROGEAR-1894) then.